components/support/utils/src/test/java/mozilla/components/support/utils/URLStringUtilsTest.kt

```diff
@@ -0,0 +1,71 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.utils
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertNotEquals
+import org.junit.Assert.assertFalse
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+import mozilla.components.support.utils.URLStringUtils.isURLLike
+import mozilla.components.support.utils.URLStringUtils.toNormalizedURL
+import mozilla.components.support.utils.URLStringUtils.isSearchTerm
+
+@RunWith(RobolectricTestRunner::class)
+class URLStringUtilsTest {
+
+    @Test
+    fun toNormalizedURL() {
+        val expectedUrl = "http://mozilla.org"
+        assertEquals(expectedUrl, toNormalizedURL("http://mozilla.org"))
+        assertEquals(expectedUrl, toNormalizedURL("  http://mozilla.org  "))
+        assertEquals(expectedUrl, toNormalizedURL("mozilla.org"))
+    }
+
+    @Test
+    fun isURLLike() {
+        assertTrue(isURLLike("mozilla.org"))
+        assertTrue(isURLLike(" mozilla.org "))
+        assertTrue(isURLLike("http://mozilla.org"))
+        assertTrue(isURLLike("https://mozilla.org"))
+        assertTrue(isURLLike("file://somefile.txt"))
+
+        assertTrue(isURLLike("http://mozilla"))
+        assertTrue(isURLLike("http://192.168.255.255"))
+        assertTrue(isURLLike("192.167.255.255"))
+        assertTrue(isURLLike("about:crashcontent"))
+        assertTrue(isURLLike(" about:crashcontent "))
+        assertTrue(isURLLike("sample:about "))
```

Comment:
Can we add one more assertion for URLs with hypens that we didn't catch in our tests from the reported Fenix bug: https://github.com/mozilla-mobile/fenix/issues/1821

`https://mozilla-mobile.org` or something.